src/main/java/com/networknt/schema/i18n/Locales.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.networknt.schema.i18n;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Locale.FilteringMode;
+import java.util.Locale.LanguageRange;
+import java.util.stream.Collectors;
+
+/**
+ * Functions for working with Locales.
+ */
+public class Locales {

```

Comment:
If I read it correctly, this class doesn't work that well with custom `MessageSource`s and you need to know what bundle is configured and use the correct list of supported Locales.

What about integrating this functionality to the `MessageSource` instead? The source should know what languages it can support and it would be responsible for choosing the `Locale`.
Then in the `ExecutionContext`/``ExecutionConfig`, you'd set only the priority list and it would get resolved through the `MessageSource` automatically.

Reply:
I prefer to separate the concerns since the use of a priority list is really an optional thing. 

Attempting to implement it in the `MessageSource` means that that it needs to keep evaluating the priority list each time for each message which won't be efficient.

If you have a custom `MessageSource` then you would know what locales you support. Then you can just use `Locales.findSupported(priorityList, mySupportedLocalesForMyMessageSource)` and just use that Locale to set in the `ExecutionConfig`.

Also you might come from a framework that already implements some of these parts, for instance Spring and then you can just use those instead, for instance Spring has a [LocaleResolver](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/servlet/LocaleResolver.html) and [MessageSource](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/MessageSource.html).

Reply:
OK, makes sense to leave the locale resolution on the client. The resolution has to be done in a bigger context (let's say of the whole app), so that the translation is consistent. 

But I still think having the list of supported languages would be better nicely coupled inside the `MessageSource`.
I think it's not ideal to rely on the fact that the developer knows what is being used where, it's better if there's a possibility to can get the information through a single API when needed. And it shouldn't matter if you use the default bundle or a custom bundle, there should be one API for one task. 
 
E.g. if this library becomes a part of something more complex, I can imaging some piece of a (utility) code that prepares some ready-made `JsonSchema` with some custom bundle. And a different piece of code for interoperability between this library and some other framework (like Spring classes that you mentioned), let's say this code is there to choose the correct locale in you app and automatically set the best candidate for your json schema validation.
These pieces can be coming from different libraries/developers and ideally, they should work together.

Reply:
The following is the typical usage

```java
SchemaValidatorsConfig config = new SchemaValidatorsConfig();
JsonSchemaFactory factory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V201909);
JsonSchema schema = factory.getSchema(source, config);

String accept = "it;q=0.9,fr;q=1.0";
Locale locale = Locales.findSupported(accept);
ExecutionContext executionContext = jsonSchema.createExecutionContext();
executionContext.getExecutionConfig().setLocale(locale);
Set<ValidationMessage> messages = jsonSchema.validate(executionContext, rootNode);
```

You can always build your own abstraction on top of this if you want to prepare a read-made `JsonSchema` with some specific configuration and some custom bundle to hide the implementation details. This will also let you switch out to a different json schema validator if need be.

Reply:
Yes, I understand that the usage, I've seen it in the new documentation, I was just suggesting to put the list of locales to the MessageSource, because it can be considered a property of a message source. Of course, one can make abstractions without that, but IMO it would be more convenient.

Reply:
@justin-tay Just want to confirm with you if this change is a breaking change. If yes, we need to make sure that we increase at least a minor number for the next release. Thanks.

Reply:
Yes, this and in particular the changes in subsequent pull requests I have made have breaking changes. Thanks for your effort in maintaining this project, it's much appreciated!